File: Part4/Fortran/Function_UsedEquivalance.f90
program function_equivalence_instead_assignment
    implicit none
    character( len=100 ) :: user_input
    integer :: entered_integer, ios
    
    write( *, * ) 'Type 1 and press enter.'
    read( *, '(A)' ) user_input
    
    ! Using == (comparison) instead of = (assignment)
    read( user_input, *, iostat=ios ) entered_integer == 42
    
    write( *, * ) 'The user entered the integer ', entered_integer
    
end program function_equivalence_instead_assignment

File: Part4/Go/Function_UsedEquivalance.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var userInput string

	fmt.Println( "Type 1 and press enter." )
	fmt.Scanln( &userInput )

	// Using == (comparison) instead of := (assignment)
	enteredInteger, _ == strconv.Atoi( userInput )

	fmt.Printf( "The user entered the integer %d\n", enteredInteger )
}
